src/app/compiler.js

```diff
@@ -24,6 +24,18 @@ function Compiler (editor, queryParams, handleGithubCall, updateFiles) {
 
   var compileTimeout = null;
 
+  executionContextEvent.register('contextChanged', this, function (context) {
```

Comment:
Hm, are you sure about this? I think these three would probably be better done in the app.